var/spack/repos/builtin/packages/netcdf-cxx/package.py

```diff
@@ -19,9 +19,20 @@ class NetcdfCxx(AutotoolsPackage):
 
     depends_on('netcdf')
 
+    variant(
+        'netcdf4', default=True, description='Compile with netCDF4 support')
+
     @property
     def libs(self):
         shared = True
         return find_libraries(
             'libnetcdf_c++', root=self.prefix, shared=shared, recursive=True
         )
+
+    def flag_handler(self, name, flags):
```

Comment:
Shouldn't this go in the configure args, does it need to be in flag_handler()?  Doesn't the configure script know about this flag?

Reply:
I am not intimately familiar with autotools so I may be may be missing something. There are some references to USE_NETCDF4 in the configure script, but I couldn't see how they would be set (there is no relevant AC_ARG_ENABLE) or how they would set the actual macro.

Reply:
The key is here:
https://www.unidata.ucar.edu/support/help/MailArchives/netcdf/msg11273.html

> If I specify -DUSE_NETCDF4 in the
> CPPFLAGS at configure time the NcFile class correctly creates Netcdf-4
> files when requested.

You need to figure out how to "specify -DUSE_NETCDF4 in the CPPFLAGS" when calling the `configure` script.  This will either involve sending the right flags to `configure` (try `-DUSE_NETCDF4`).  Or it will involve implementing (I think) a `setup_build_envrionment()` method in Spack.  The former solution is preferable, if it works.  Either way, use of `flag_handler()` for this purpose is not really the way things were designed to work.

Reply:
I've updated the package to use `setup_environment()`. I never found any way to do this via the configure script, after looking through the script and trying:

    ./configure --enable-netcdf4
    ./configure -DUSE_NETCDF4
    USE_NETCDF4=true ./configure
    USE_NETCDF4_TRUE=true ./configure
    CPPFLAGS=-DUSE_NETCDF4 ./configure

The last one of course worked, which is what `setup_environment()` now does.